tensorflow_model_optimization/python/core/sparsity_tf2/pruner_test.py

```diff
@@ -0,0 +1,493 @@
+  #       K.get_session().run(assign_add(self.global_step, 1))
+
+  #     non_zero_count.append(np.count_nonzero(K.get_value(weight)))
+
+  #   # Weights pruned at steps 1,3,5
+  #   expected_non_zero_count = [100, 90, 90, 70, 70, 50, 50, 50, 50, 50]
+  #   self.assertAllEqual(expected_non_zero_count, non_zero_count)
+
+
+  def testSaveOGSlotInit(self):
+    initial_weight = tf.Variable(np.linspace(1.0, 100.0, 100), name="weights")
+    weight_dtype = initial_weight.dtype.base_dtype
+    save_round = 0
+    n_rounds = 5
+    end_itr = 100
+    frequency, prune_pcent_per_round = get_lth_sparsity(save_round, n_rounds, self.target_sparsity, end_itr)
+    pruning_schedule = make_pruning_schedule(prune_pcent_per_round, 0, end_itr, frequency)
+
+    self.initialize()
+    
+    p = pruner.LTHPruner(
+        pruning_schedule=pruning_schedule,
+        save_schedule=self.save_init,
+        block_size=self.block_size,
+        block_pooling_type=self.block_pooling_type)
+
+    optimizer = self.dummy_optimizer
+    optimizer.iterations.assign(tf.cast(self.training_step_fn(), tf.int64))
+    init_weights_before_pruning = initial_weight.read_value()
+
+    # TODO: before creating slots, should exist
+    p.create_slots(optimizer, initial_weight)
+    initial_weight.assign(tf.math.add(initial_weight, sample_noise(0)))
+    p.prune(optimizer, initial_weight, self.grad(initial_weight))
+    og_initialization_slot = optimizer.get_slot(initial_weight, "original_initialization")
+    self.assertAllEqual(og_initialization_slot, init_weights_before_pruning)
+
+
+  # def testSaveWeightsItrK(self):
+  #   initial_weight = tf.Variable(np.linspace(1.0, 100.0, 100), name="weights")
+  #   weight_dtype = initial_weight.dtype.base_dtype
+
+  #   save_round = 5
+  #   n_rounds = 20
+  #   end_itr = 100
+  #   frequency, prune_pcent_per_round = get_lth_sparsity(save_round, n_rounds, self.target_sparsity, end_itr)
+  #   pruning_schedule = make_pruning_schedule(prune_pcent_per_round, save_round, end_itr, frequency)
+  #   # NOTE: save_round > start = 0, should not prune until save round is passed
+
+  #   self.initialize()
+    
+  #   p = pruner.LTHPruner(
+  #       pruning_schedule=pruning_schedule,
+  #       save_schedule=self.save_init,
+  #       block_size=self.block_size,
+  #       block_pooling_type=self.block_pooling_type)
+
+  #   optimizer = self.dummy_optimizer
+  #   optimizer.iterations.assign(tf.cast(self.training_step_fn(), tf.int64))
+  #   expected_saved_og_initialization = None
+
+  #   p.create_slots(optimizer, initial_weight)
+  #   for i in range(9): # this should save and not update, assumes pruning works correctly as per above tests
+  #     # perturb weights
+  #     initial_weight = tf.math.add(initial_weight, sample_noise(i))
+  #     p.prune(optimizer, initial_weight, self.grad(initial_weight))
+  #     if i == save_round:
+  #       expected_saved_og_initialization = optimizer.get_slot(initial_weight, "original_initialization").read_value()
```

Comment:
I believe you mean to capture the weight here to compare against, not the original_initialization.
`expected_saved_initialization = weight.read_value()`

Reply:
(this applies to all places you do this in the tests)